configuration/rules-nools.md

```diff
@@ -0,0 +1,154 @@
+### Nools overview
+
+[Nools](https://github.com/noolsjs/nools#usage) is a rete based engine written in javascript.
+
+Understanding the Rete algorithm will be helpful in writing optimized rules for task and target (or other event) generation.
+
+> The genius of the approach is to keep in memory the “status” of individual condition evaluations removing the need for constant calculations.  As a result, rules project that are by definition pattern-matching intensive can be evaluated much faster, especially as new facts need to be propagated, than using a systematic approach.
+
+Read up on how Rete works [here](https://www.sparklinglogic.com/rete-algorithm-demystified-part-2/) and [here](http://www.jbug.jp/trans/jboss-rules3.0.2/ja/html/ch01s04.html) 
+
+
+The basic structure of your rules.nools.js should look something like this:
+
+
+```
+define Target {
+    _id: null,
+    deleted: null,
+    type: null,
+    pass: null,
+    date: null
+}
+
+define Contact {
+    contact: null,
+    reports: null
+}
+
+define Task {
+    _id: null,
+    deleted: null,
+    doc: null,
+    contact: null,
+    icon: null,
+    date: null,
+    title: null,
+    fields: null,
+    resolved: null,
+    priority: null,
+    priorityLabel: null,
+    reports: null,
+    actions: null
+}
+rule GenerateClinicEvents {
+    when {
+
+    }
+    then {
+
+    }
+}
+```
+
+`Target`, `Contact` and `Task` are object classes.
+
+The Rete algorithm can be broken into the rule compilation and runtime execution.
+
+The compilation algorithm describes how the Rules in the Production Memory generate and efficient discrimination network. A discrimination network is used to filter data.
```

Comment:
First sentence has slight grammatical error - maybe it should say "generate an efficient discrimination network"?